File: backend/app/schemas/provider.py
from pydantic import BaseModel
from typing import Optional


class Address(BaseModel):
    street: str
    city: str
    state: str
    zip: str


class Provider(BaseModel):
    npi: str
    name: str
    provider_type: str
    specialty: str
    facility_type: str
    address: Address
    phone: str
    accepts_medicaid: bool
    accepts_medicare: bool
    sliding_scale: bool
    distance_miles: Optional[float] = None


class ProcedurePrice(BaseModel):
    cpt_code: str
    procedure_name: str
    cash_price: float
    medicare_rate: Optional[float] = None
    estimated_insured_price: Optional[float] = None
    price_range_low: float
    price_range_high: float


class ProviderDetail(Provider):
    procedure_prices: list[ProcedurePrice]
    hours: str
    website: Optional[str] = None
    rating: Optional[float] = None

File: frontend/src/types/provider.ts
export interface Provider {
  npi: string;
  name: string;
  provider_type: "individual" | "organization";
  specialty: string;
  facility_type: string;
  address: {
    street: string;
    city: string;
    state: string;
    zip: string;
  };
  phone: string;
  accepts_medicaid: boolean;
  accepts_medicare: boolean;
  sliding_scale: boolean;
  distance_miles?: number;
}

export interface ProcedurePrice {
  cpt_code: string;
  procedure_name: string;
  cash_price: number;
  medicare_rate: number | null;
  estimated_insured_price: number | null;
  price_range_low: number;
  price_range_high: number;
}

export interface ProviderDetail extends Provider {
  procedure_prices: ProcedurePrice[];
  hours: string;
  website?: string;
  rating?: number;
}
